### PR TITLE
WIP: wxwidgets: fix build on darwin

### PR DESCRIPTION
**Pull request by willcohen**
###### Motivation for this change

darwin packaging depending on wxwidgets fail due to #126101. Rather than forcing all upstream dependencies not needing webkitgtk to exclude it, it may be simpler to exclude here until fixed. Additionally, AVFoundation appears to be needed on my x86 machine to complete a build.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/wxwidgets/3.0/default.nix [hunk 1]
```diff
@@ -8,7 +8,7 @@
 , compat24 ? false, compat26 ? true, unicode ? true
 , withGtk2 ? true
 , withWebKit ? false, webkitgtk
```
Comment (veprbl) [inline pkgs/development/libraries/wxwidgets/3.0/default.nix:10]:
```suggestion
, withWebKit ? !stdenv.isDarwin, webkitgtk # Disable webkitgtk on darwin pending #126101
```
```suggestion
, withWebKit ? false, webkitgtk
```

File: pkgs/development/libraries/wxwidgets/3.0/default.nix [hunk 2]
```diff
@@ -33,8 +33,8 @@ stdenv.mkDerivation rec {
   ] ++ optional withGtk2 gtk2
     ++ optional (!withGtk2) gtk3
     ++ optional withMesa libGLU
-    ++ optional withWebKit webkitgtk
-    ++ optionals stdenv.isDarwin [ setfile Carbon Cocoa Kernel QTKit ];
+    ++ optional (withWebKit && !stdenv.isDarwin) webkitgtk # Disable webkitgtk on darwin pending #126101
```
Comment (veprbl) [inline pkgs/development/libraries/wxwidgets/3.0/default.nix:36]:
```suggestion
    ++ optional withWebKit webkitgtk
```
```suggestion
    ++ optional (withWebKit && !stdenv.isDarwin) webkitgtk # Disable webkitgtk on darwin pending #126101
```

Comment (willcohen) [reply pkgs/development/libraries/wxwidgets/3.0/default.nix]: Much cleaner, thanks! Applied below.

**Comment by willcohen** [comment]
@peterhoeg while in some sense the `wxGTK30: You cannot enable withWebKit when using withGtk2.` failure is irrelevant to darwin and this particular PR since webkit has to be entirely disabled, should webkit be disabled by default for all while #73145 is pending?

**Comment by willcohen** [comment]
Additionally, the reason I only enabled darwin builds on `wxGTK30` and not `wxGTK31` is that for 3.1, the build fails on my machine with because of https://github.com/NixOS/nixpkgs/blob/c81e01e9b9ce248cfd15d28926fd286efe2374f4/pkgs/development/libraries/wxwidgets/3.1/default.nix#L90, with ``--enable-universal-binaries`` being considered an invalid option, while it worked fine for 3.0.

File: pkgs/development/libraries/wxwidgets/3.0/default.nix [hunk 3]
```diff
@@ -7,8 +7,8 @@
 , libGLU, libGL
 , compat24 ? false, compat26 ? true, unicode ? true
 , withGtk2 ? true
-, withWebKit ? false, webkitgtk
-, AGL, Carbon, Cocoa, Kernel, QTKit
+, withWebKit ? !stdenv.isDarwin, webkitgtk  # Disable webkitgtk on darwin pending #126101
```
Comment (veprbl) [inline pkgs/development/libraries/wxwidgets/3.0/default.nix:10]: Actually, the original value was already false, so setting it to `!stdenv.isDarwin` would enable it on non-darwin. And I don't see any override in all-packages.nix
```suggestion
, withWebKit ? false, webkitgtk
```

**Comment by veprbl** [comment]
> Additionally, the reason I only enabled darwin builds on `wxGTK30` and not `wxGTK31` is that for 3.1, the build fails on my machine with because of
> 
> https://github.com/NixOS/nixpkgs/blob/c81e01e9b9ce248cfd15d28926fd286efe2374f4/pkgs/development/libraries/wxwidgets/3.1/default.nix#L90
> 
> , with `--enable-universal-binaries` being considered an invalid option, while it worked fine for 3.0.

Usually autotools only prints warnings on seeing unknown options. What is the error that you see?

**Comment by veprbl** [comment]
Also, just to make sure, you really want `wxGTK` running on darwin? or `wxmac` would work for your case too?

**Comment by veprbl** [comment]
> configure: error: unrecognized options: --enable-universal-binaries

Well, then this option should not passed.

> I think I may have missed this very obvious solution before embarking on this workaround. I'm not actually sure that GTK is necessary for the use cases I'm envisioning.

To avoid such confusion, we probably need to add a generic `wxWidgets` implementation that would point to the one most proliferent on the current platform.

**Comment by willcohen** [comment]
In the interest of not over-complicating something that isn't particularly broken or creating yet more churn over something which probably isn't top of the list for ways to make the ecosystem less confusing for newbies, I'm inclined to simply close this PR and leave it as a bit of keyword-appropriate guidance for someone who finds themselves in a similar situation. `wxmac` solves all the needs I have!

**Pull request closed without merging**